tools/condition.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2022, seL4 Project a Series of LF Projects, LLC
+#
+# SPDX-License-Identifier: BSD-2-Clause or GPL-2.0-only
+#
+
+def remove_ws_comments(node):
+    for child in list(node.childNodes):
+        if child.nodeType == node.COMMENT_NODE or (child.nodeType == node.TEXT_NODE and len(child.data.strip()) == 0):
+            node.removeChild(child)
+
+
+def condition_to_cpp(conditions):
+    n = len(conditions)
+    assert n <= 1
+    if n == 0:
+        return ""
+    else:
+        remove_ws_comments(conditions[0])
+        children = conditions[0].childNodes
+        if children == None or len(children) == 0:
+            return ""
+
+        assert len(children) == 1
+
+        def helper(expr):
+            remove_ws_comments(expr)
+            if expr.tagName == "config":
+                return "defined({})".format(expr.getAttribute("var"))
+            elif expr.tagName == "not":
+                child = expr.firstChild
+                if child.tagName == "and" or child.tagName == "or":
+                    return "!({})".format(helper(child))
+                else:
```

Comment:
What is `child.tagName` in this case? Could avoid unconditional else and reject anything unknown here to ensure this raises an exception if somebody every extends this mechanism and forgets to update this